packages/browser/__tests__/Session.spec.ts

```diff
@@ -487,6 +487,28 @@ describe("Session", () => {
         mySession.handleIncomingRedirect("https://some.url")
       ).resolves.not.toThrow();
     });
+
+    it("sets the appropriate information before calling the callback", async () => {
+      let hasBeenCalled = false;
+      const clientAuthentication = mockClientAuthentication();
+      clientAuthentication.handleIncomingRedirect = jest.fn(
+        async (_url: string) => {
+          return {
+            isLoggedIn: true,
+            sessionId: "a session ID",
+            webId: "https://some.webid#them",
+          };
+        }
+      );
+      const mySession = new Session({ clientAuthentication });
+      const myCallback = (): void => {
+        hasBeenCalled = true;
+        expect(mySession.info.webId).toBe("https://some.webid#them");
+      };
+      mySession.onLogin(myCallback);
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(hasBeenCalled).toEqual(true);
```

Comment:
If you make this a mock function you don't need to manually track calls, I think. Also, be sure to add an `expect.assertions(2)` to make sure the conditional `expect` is actually called.

```suggestion
      const myCallback = jest.fn((): void => {
        expect(mySession.info.webId).toBe("https://some.webid#them");
      });
      mySession.onLogin(myCallback);
      await mySession.handleIncomingRedirect("https://some.url");
      expect(myCallback).toHaveBeenCalledTimes(1);
```

Reply:
`expect.assertions()` is added in 059619d999004cd0bae8abeb5ac91d0fba343a80